### PR TITLE
Fix external provider for openshift-ci usage

### PR DESCRIPTION
**Pull request by dhiller**
Fix needed to use the external provider, otherwise it won't find the configuration.

Signed-off-by: Daniel Hiller <daniel.hiller.1972@gmail.com>

### CODE REVIEW

File: cluster-up/cluster/external/provider.sh [hunk 1]
```diff
@@ -5,11 +5,17 @@ function _kubectl() {
...
 function prepare_config() {
-    cat >hack/config-provider-external.sh <<EOF
-docker_tag=devel
-docker_prefix=${DOCKER_PREFIX}
-manifest_docker_prefix=${DOCKER_PREFIX}
-image_pull_policy=${IMAGE_PULL_POLICY:-Always}
+    BASE_PATH=${KUBEVIRTCI_CONFIG_PATH:-$PWD}
+
+    # required for running tests within openshift ci, otherwise tests don't find the configuration
+    ln -f -s ${KUBECONFIG:-$HOME/.kube/config} ${BASE_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig
```
Comment (danielBelenky) [inline cluster-up/cluster/external/provider.sh]: Could you wrap the expressions here with quotes? Though unlikely, but if KUBECONFIG will have a space in its path this command will fail.

I'd suggest quoting always unless you explicitly want bash to split your expression

Comment (dhiller) [reply cluster-up/cluster/external/provider.sh]: Yeah, thought so too. I'd rather fix all `shellcheck` warnings but didn't want to go too far here ;-)

Comment (dhiller) [reply cluster-up/cluster/external/provider.sh]: Done.

File: cluster-up/cluster/external/provider.sh [hunk 2]
```diff
@@ -5,19 +5,24 @@ function _kubectl() {
...
 function prepare_config() {
-    cat >hack/config-provider-external.sh <<EOF
-docker_tag=devel
-docker_prefix=${DOCKER_PREFIX}
-manifest_docker_prefix=${DOCKER_PREFIX}
-image_pull_policy=${IMAGE_PULL_POLICY:-Always}
+    BASE_PATH=${KUBEVIRTCI_CONFIG_PATH:-$PWD}
+
+    # required for running tests within openshift ci, otherwise tests don't find the configuration
+    ln -f -s "${KUBECONFIG:-$HOME/.kube/config}" "${BASE_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig"
```
Comment (rmohr) [inline cluster-up/cluster/external/provider.sh]: I am not sure what you mean by that. You can set the config location on the tests binary. Is that what you try to solve here?

It probably does not work right now, but the provider is supposed to just find the config at a default location or in KUBECONFIG.

Comment (dhiller) [reply cluster-up/cluster/external/provider.sh]: Yes, that is exactly what I'm trying to solve here. Within openshift-ci `KUBECONFIG` is set to `/tmp/admin.kubeconfig`. Unfortunately the tests won't find the configuration unless I symlink to the target configuration.

I agree that this is a symptom, not the cause. Opened #202 for this.

**Comment by dhiller** [comment]
/cc @rmohr @danielBelenky 

can you please re-review?

**Comment by rmohr** [comment]
/lgtm
/approve
/hold

let's reverify that it works and take it in.

File: cluster-up/cluster/external/provider.sh [hunk 3]
```diff
@@ -5,19 +5,33 @@ function _kubectl() {
...
+    if [ -z "${KUBECONFIG}" ]; then
+        echo "KUBECONFIG is not set!"
+        exit 1
+    fi
+
+    PROVIDER_CONFIG_FILE_PATH="${BASE_PATH}/$KUBEVIRT_PROVIDER/config-provider-$KUBEVIRT_PROVIDER.sh"
+
+    cat > "$PROVIDER_CONFIG_FILE_PATH" <<EOF
+kubeconfig=\${KUBECONFIG}
+docker_tag=\${DOCKER_TAG}
```
Comment (rmohr) [inline cluster-up/cluster/external/provider.sh]: should we fall back to `devel` here?

**Pull request merged**